angular/src/lib/select/select.component.ts

```diff
@@ -1,41 +1,36 @@
 import {
+  AfterContentChecked,
+  ChangeDetectorRef,
   Component,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  HostBinding,
+  HostListener,
   Input,
-  ViewChild,
-  AfterContentChecked,
-  ContentChildren,
-  QueryList,
-  AfterContentInit,
-  OnInit,
   NgZone,
+  OnDestroy,
+  OnInit,
   Output,
-  EventEmitter,
-  HostListener,
-  ElementRef,
-  HostBinding
+  TemplateRef,
+  ViewChild
 } from '@angular/core';
-import {SelectionModel} from '@angular/cdk/collections';
-
+import { animate, AnimationEvent, state, style, transition, trigger } from '@angular/animations';
 import uniqueId from 'lodash-es/uniqueId';
 import { ButtonComponent } from '../button/button.component';
-import { ListItemComponent, OptionSelectionChange } from '../list-item/list-item.component';
+import { Subscription } from 'rxjs';
+import { FilterUtils } from '../utils/filterUtils/filters';
+import findIndex from 'lodash-es/findIndex';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+import { SelectService } from './select.service';
+
+const SELECT_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  // tslint:disable-next-line: no-use-before-declare
+  useExisting: forwardRef(() => SelectComponent),
```

Comment:
it always concerns me when we disable a tslint line .. what is the reasoning here?

Reply:
Provides declaration of the SELECT_VALUE_ACCESSOR (ng_value_accessor) variable to the providers [] of the Select component. All components using the ng_value_accessor have the same.